core/src/banking_stage/latest_unprocessed_votes.rs

```diff
@@ -283,6 +300,48 @@ impl LatestUnprocessedVotes {
             .and_then(|l| l.read().unwrap().timestamp())
     }
 
+    #[cfg(test)]
+    pub(crate) fn set_staked_nodes(&self, staked_nodes: &[Pubkey]) {
+        let staked_nodes: HashMap<Pubkey, u64> =
+            staked_nodes.iter().map(|pk| (*pk, 1u64)).collect();
+        *self.cached_staked_nodes.write().unwrap() = Arc::new(staked_nodes);
+    }
+
+    fn weighted_random_order_by_stake(&self) -> impl Iterator<Item = Pubkey> {
```

Comment:
It's hard to tell what changed with this function since it was moved. What's the intention of the changes here?

Reply:
the functionality did not change, it uses the stake info from the struct rather than having it be passed in through the bank

Reply:
> uses the stake info from the struct rather than having it be passed in through the bank

Is this because the stakes in the bank might lag the stakes that we snapshot and pass in? Or vice versa?